STACpopulator/implementations/Ouranos_CMIP6-CORDEX/add_CORDEX6.py

```diff
@@ -0,0 +1,75 @@
+import argparse
+import json
+import logging
+import os
+import sys
+from typing import Any, MutableMapping, Optional, Union
+
+from pystac import STACValidationError
+from pystac.extensions.datacube import DatacubeExtension
+from requests.sessions import Session
+
+from STACpopulator.requests import add_request_options, apply_request_options
+from STACpopulator.extensions.cmip6 import CMIP6Helper, CMIP6Properties
+from STACpopulator.extensions.datacube import DataCubeHelper
+from STACpopulator.extensions.thredds import THREDDSExtension, THREDDSHelper
+from STACpopulator.input import ErrorLoader, GenericLoader, THREDDSLoader
+from STACpopulator.models import GeoJSONPolygon
+from STACpopulator.populator_base import STACpopulatorBase
+
+LOGGER = logging.getLogger(__name__)
+import argparse
+from typing import Any
+from STACpopulator.populator_base import STACpopulatorBase
+from STACpopulator.extensions.cordex6 import Cordex6DataModel
+
+
+class CORDEX_STAC_Populator(STACpopulatorBase):
+    data_model = Cordex6DataModel
+    item_geometry_model = None  # Unnecessary, but kept for consistency
```

Comment:
This is defined for the `CMIP6populator`: 

```python
class CMIP6populator(STACpopulatorBase):
    item_properties_model = CMIP6Properties
    item_geometry_model = GeoJSONPolygon
```

And `data_model = Cordex6DataModel` basically offers: 

```python
Cordex6DataModel.properties == CordexCmip6  #  -> just like CMIP6Properties
Cordex6DataModel -> THREDDSCatalogDataModel.geometry  # -> just like item_geometry_model
```

I'm wondering if there's any duplication of the intended use of these properties?

Reply:
Yes, because I didn't want to break the CMIP extension and implementation just yet. My idea was to try to generalize the CORDEX example, get a sense of where this is going, and once we're happy, then bring the changes to CMIP6.